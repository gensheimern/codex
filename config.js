module.exports = {
	DB_HOST: process.env.DB_HOST || '159.65.115.221',
	DB_USER: process.env.DB_USER || 'root',
	DB_PASS: process.env.DB_PASS || 'teamcodex',
	DB_NAME: process.env.DB_NAME || 'lunch_planner',
	JWT_SECRET: process.env.JWT_SECRET || 'secret',
	MAIL_HOST: process.env.MAIL_HOST || '159.65.115.221',
	MAIL_USER: process.env.MAIL_USER || 'support@codex-team.de',
	MAIL_PASS: process.env.MAIL_PASS || 'teamcodex123',
};
