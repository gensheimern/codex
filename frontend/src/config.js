const config = {
	debug: process.env.REACT_APP_DEBUG === 'TRUE',
	apiPath: process.env.REACT_APP_API_PATH,
	wsPath: process.env.REACT_APP_WS_PATH,
	basePath: process.env.REACT_APP_BASE_PATH,
}

export default config;