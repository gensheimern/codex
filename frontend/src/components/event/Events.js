import React from 'react';
import EventItem from './EventItem';
import AddButton from './AddButton';
import config from '../../config';
import MediaQuery from 'react-responsive';

export default class Events extends React.Component {

	constructor(props) {
		super(props);

		this.state = {
			events:[],
			loaded: false,
			error: null,
		};
	}

	componentDidMount() {
		this.loadActivityData();
	}

	loadActivityData() {
		this.setState({
			error: null,
			loaded: false,
		});

		fetch(config.apiPath + "/activity", {
			method: 'GET',
			headers: {
				'Content-Type': 'application/json',
				'X-Access-Token': localStorage.getItem('apiToken')
			}
		})
		.then((res) => {
			if (!res.ok) {
				throw new Error("Request failed.");
		  	} else if (res.status !== 200) {
				throw new Error("Forbidden");
			}

			return res;
		})
		.then(res => res.json())
		.then(res => {
			this.setState({
				events: res,
				loaded: true,
		  });

		})
		.catch((err) => {
			this.setState({
				error: 'An Error occured.',
			});
		});
	}

	render() {
		if (this.state.error) {
			return (<p>{this.state.error}</p>);
		}

		if(!this.state.loaded) {
			return (<p>Loading</p>);
		}

		return (
			<React.Fragment>
				<div>
				<MediaQuery query="(min-device-width: 768px)">
					{(matches) => {
						if (matches) {
							return <div>
							<div className="feedInfo" style={{paddingLeft: "2%"}}>
								<h3>PUBLIC</h3>
								<p>12 | Welcome to the public channel of VSF Experts. You can find all your colleagues here.</p>
							</div>
							<div className="addButton">
							<hr />
							<AddButton />
							</div>
							 </div>
						}
						else {
							return null
						}
					}
			}
			</MediaQuery>
				<div className="feed">
					{ this.state.events.map(event => (<EventItem key={event.id} event={event} />)) }
				</div>
				</div>
			</React.Fragment>
		);

	}
}
