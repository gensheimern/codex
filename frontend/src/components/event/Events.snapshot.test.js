import React from 'react';
import renderer from'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Events from './Events';
import config from '../../config';

const fetchMock = require('fetch-mock');

beforeEach(() => {
    window.localStorage = {getItem: () => ""};
    fetchMock.get(config.apiPath +'/activity/5/message', 
    [ { id: 6,
        time: new Date('12.12.2018'),
        content: "",
        author:{
            id:7,
            firstName: "",
            name: "",
            email: "",
            img: "",
        }
    }]);
    fetchMock.get( config.apiPath +'/user/me',{
        id:7,
        firstName: "Max",
        name: "Mustermann",
        email: "max.mustermann@gmail.com",
        image: "",
    });
    fetchMock.get(config.apiPath +'/activity',{
        id:8,
		description: "",
		name: "",
		place: "",
		time: new Date('01.06.2018'),
		event: true,
		private: true,
		banner: "",
		maxParticipants: "",
		host: {
			id:7,
            firstName: "Max",
            name: "Mustermann",
            email: "max.mustermann@gmail.com",
            image: "",
            me: true,
		},
    });
  });

  afterEach(() => {
   fetchMock.restore();
  });

describe('Events Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <MuiThemeProvider>
            <Events/>
            </MuiThemeProvider>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});