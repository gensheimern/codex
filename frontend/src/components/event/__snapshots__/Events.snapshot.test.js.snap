// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Snapshot renders 1`] = `
<p>
  Loading
</p>
`;
